cli/src/commands/runTests.js

```diff
@@ -147,160 +148,182 @@ function printSkipMessage(flowVersion, githubUrl) {
   );
 }
 
+// Once we've fetched flow bin releases they will be cached
+// For subsequent tests within the same test run.
+let _flowReleases = [];
+
 /**
  * Memoized function that queries the GitHub releases for Flow, downloads the
  * zip for each version, extracts the zip, and moves the binary to
  * TEST_BIN/flow-vXXX for use later when running tests.
  */
-let _flowBinVersionPromise = null;
 async function getOrderedFlowBinVersions(
-  numberOfReleases: number = 15,
+  numberOfReleases: number,
+  flowVersion: FlowVersion,
 ): Promise<Array<string>> {
-  if (_flowBinVersionPromise !== null) {
-    return _flowBinVersionPromise;
-  }
-  return (_flowBinVersionPromise = (async function() {
```

Comment:
This blocks main change is the removal of an anonymous function causing a big diff, main changes are addition of calling `selectFlowTestVersions` and the while loop that pulls all release versions